### PR TITLE
Support Coda with Tweedle curve and SchnorrPoseidon signatures

### PR DESCRIPTION
**Pull request by lk86**
Fixes # .

### Motivation

Enables support for Coda Protocol's rosetta implementation, specifically the curve type and signature scheme used in Coda.

### Solution

Adds Coda's curve type (tweedle) and signature scheme (schnorr_poseidon) to the specification.

### CODE REVIEW

**Comment by patrick-ogrady** [comment]
👋 @lk86 ... you'll need to run `make gen` to fix the `check-gen` test.

File: models/CurveType.yaml [hunk 1]
```diff
@@ -13,11 +13,13 @@
 # limitations under the License.
 
 description: |
-  CurveType is the type of cryptographic curve associated with a PublicKey.
```
Comment (patrick-ogrady) [inline models/CurveType.yaml:16]: ❤️

**Comment by patrick-ogrady** [comment]
Looks like you'll need to rebase onto master here too @lk86 ... we added `secp256r1` while you were working on this. (https://github.com/coinbase/rosetta-specifications/pull/39)

**Comment by lk86** [comment]
I made sort of a mess of the commit history trying to rebase across forks but the changes here should now be properly in sync with master.

**Comment by patrick-ogrady** [comment]
Our team is reviewing the docs you linked @lk86, will get back to you with any comments or approve shortly.

**Comment by patrick-ogrady** [comment]
Would you mind adding your new `CurveType` and `SignatureType` to the README: https://github.com/coinbase/rosetta-specifications#writing-a-construction-api-implementation

**Comment by lk86** [comment]
> Would you mind adding your new `CurveType` and `SignatureType` to the README: https://github.com/coinbase/rosetta-specifications#writing-a-construction-api-implementation
> 
> We would also appreciate it if you could elaborate on the challenge value in the signature description + README (i.e. what you have here: https://github.com/CodaProtocol/signer-reference/blob/c4dee525d6a8ff947ab10bd849e74936d29e1212/schnorr.ml#L92). Just putting something like `hash(R || P || m)` (replacing the values inside the hash with what is used in Coda) should be sufficient.

I've updated tweedle and schnorr_poseidon in the readme, and added a high-level calculation of the schnorr challenge we use (`hash(1st pk || 2nd pk || r)`).

**Comment by patrick-ogrady** [comment]
Looks good @lk86 ... once you run `make gen`, should be good!

**Comment by lk86** [comment]
> Looks good @lk86 ... once you run `make gen`, should be good!

whoops, should be all set now!

**Review by patrick-ogrady — APPROVED** [review]
👍

**Pull request merged**